src/mysql-runner/StatementReader.cs

```diff
@@ -99,15 +134,21 @@ private bool IsTerminated(List<string> parts)
                 return false;
             }
 
+            // Prevent double counting in case nothing was added to the parts
+            if (_lastBlockStartIndex == parts.Count())
```

Comment:
minor: braces style: prefer blocks to always have surrounding braces, ie

```csharp
if (_lastBlockStartIndex == parts.Count)
{
  return false;
}
```

also minor: `List<T>` has a `Count` property - rather use that than calling into LINQ's `.Count()`

Reply:
Fixed